juju/juju
# caas upgrade worker and k8s provider support for operator upgrades

## Description of change

Add in the backend for `juju upgrade-controller` for k8s controllers.
There's a new caas upgrade worker based off the iaas upgrade worker. The worker simply patches the image patch of the operator image to match the desired version.

Due to a bug in pod restart, the upgrade is kicked off but the pod can't come up again. This is being worked on.

## QA steps

bootstrap into k8s
upload to docker an operator image tagged with a newer version
run upgrade-controller
use kubectl to see that the controller pod restarts with the newer version image

ensure upgrade iaas still works
